### PR TITLE
feat: [SSCA-1044] use OPA policy picker behind FF

### PR DESCRIPTION
**Pull request by udayvunnam**
### Summary

**With FF off** 

show only the existing behaviour of filestore picker, no toggle to select OPA

**With FF on** 
show a toggle to switch between OPA and filestore

- if there's existing filestore, toggle off by default
- else toggle on by default and show the OPA picker

**Update CISscaOrchestration -> SscaOrchestration as this will common for both CI and CD going forward**

FF integration SSCA-1043
#### Screenshots

**current**
image

**new**
image

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

**Comment by udayvunnam** [comment]
retrigger messagecheck

**Comment by udayvunnam** [comment]
trigger messagecheck

File: src/modules/75-ssca/components/PipelineSteps/__tests__/SscaEnforcementStep.test.tsx [hunk 1]
```diff
@@ -90,12 +111,72 @@ const fixedValues = {
...
+    expect(screen.queryByText('ssca.useOpaPolicy')).not.toBeInTheDocument()
+    expect(screen.queryByText('common.policiesSets.addOrModifyPolicySet')).not.toBeInTheDocument()
+  })
+
+  test('edit view as new step | OPA policy FF On', async () => {
+    const initialValues = cloneDeep(fixedValues)
+    initialValues.spec = {
+      ...commonDefaultEnforcementSpecValues,
+      ...ciSpecValues
+    } as any
```
Comment (bhavya-sinha) [inline src/modules/75-ssca/components/PipelineSteps/__tests__/SscaEnforcementStep.test.tsx:137]: Can we have proper type

File: src/modules/75-ssca/components/PipelineSteps/__tests__/SscaEnforcementStep.test.tsx [hunk 2]
```diff
@@ -7,14 +7,35 @@
...
 import { StepType } from '@pipeline/components/PipelineSteps/PipelineStepInterface'
 import { factory, TestStepWidget } from '@pipeline/components/PipelineSteps/Steps/__tests__/StepTestUtil'
 import { doConfigureOptionsTesting, queryByNameAttribute } from '@common/utils/testUtils'
-import { CiSscaEnforcementStep } from '../CiSscaEnforcementStep/CiSscaEnforcementStep'
-import { SscaCiEnforcementStepData } from '../common/types'
+import { SscaEnforcementStep } from '../SscaEnforcementStep/SscaEnforcementStep'
+import { SscaEnforcementStepData } from '../common/types'
+import { ciSpecValues, commonDefaultEnforcementSpecValues } from '../common/utils'
+
+jest.mock('@connectors/components/ConnectorReferenceField/FormMultiTypeConnectorField', () => ({
```
Comment (chetanvekariya-harness) [inline src/modules/75-ssca/components/PipelineSteps/__tests__/SscaEnforcementStep.test.tsx:21]: We should not mock components. These components are used at many places. You might be facing issues with api mocks and that is why you have mocked components. By mocking couple of APIs, you should be fine. You may check how it is done in any infra spec test files. For eg, AwsLambdaInfraSpec

File: src/modules/75-ssca/components/PipelineSteps/common/SscaEnforcementStepInputSet.tsx [hunk 3]
```diff
@@ -67,6 +68,19 @@ export default function SscaEnforcementStepInputSet(
           />
         </div>
       )}
+
+      {isValueRuntimeInput(get(template, 'spec.policy.policySets', '')) && (
```
Comment (chetanvekariya-harness) [inline src/modules/75-ssca/components/PipelineSteps/common/SscaEnforcementStepInputSet.tsx:72]: Is this covered with test case?

Comment (udayvunnam) [reply src/modules/75-ssca/components/PipelineSteps/common/SscaEnforcementStepInputSet.tsx:72]: added now

File: src/modules/75-ssca/components/PipelineSteps/__tests__/SscaEnforcementStep.test.tsx (hunk 2 above)
Comment (udayvunnam) [reply src/modules/75-ssca/components/PipelineSteps/__tests__/SscaEnforcementStep.test.tsx]: copied this from elsewhere to mock this component. I see AwsLambdaInfraSpec has clean mocking. updated. Thank you :)

**Comment by bhavya-sinha** [comment]
Jest is failing

File: src/modules/75-ssca/components/PipelineSteps/common/SscaEnforcementStepEdit.tsx [hunk 4]
```diff
@@ -79,16 +82,26 @@ const SscaEnforcementStepEdit = <T,>(
...
   const gitScope = useGitScope()
 
+  const _initialValues = getInitialValuesInCorrectFormat<T, T>(initialValues, transformValuesFieldsConfig(stepType))
+
+  if (
+    SSCA_ENFORCEMENT_WITH_BOTH_NATIVE_AND_OPA_POLICIES_ENABLED &&
+    !get(_initialValues, 'spec.policy.store.spec.file')
+  ) {
+    set(_initialValues as any, 'spec.policy.opa', true)
+    set(_initialValues as any, 'spec.policy.store', undefined)
```
Comment (bhavya-sinha) [inline src/modules/75-ssca/components/PipelineSteps/common/SscaEnforcementStepEdit.tsx:92]: here too

**Pull request merged**